### PR TITLE
FlatGeobuf: metadata extensions

### PR DESCRIPTION
**Pull request by bjornharrtell**
ref https://github.com/bjornharrtell/flatgeobuf/pull/75

### CODE REVIEW

File: gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp [hunk 1]
```diff
@@ -239,11 +253,27 @@ void OGRFlatGeobufLayer::readColumns()
...
+        const auto ogrType = toOGRFieldType(column->type(), eSubType);
+        OGRFieldDefn field(name, ogrType);
         field.SetSubType(eSubType);
+        field.SetAlternativeName(title);
+        if (width != 1 && type != ColumnType::Float && type != ColumnType::Double)
+            field.SetWidth(width);
+        if (precision != -1)
+            field.SetWidth(precision);
+        field.SetPrecision(scale != -1 ? scale : 0);
+        field.SetPrecision(scale != -1 ? scale : 0);
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp:274]: duplicated
```suggestion
```

File: gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp [hunk 2]
```diff
@@ -239,11 +253,27 @@ void OGRFlatGeobufLayer::readColumns()
...
+        const auto nullable = column->nullable();
+        const auto unique = column->unique();
         OGRFieldSubType eSubType = OFSTNone;
-        const auto type = toOGRFieldType(column->type(), eSubType);
-        OGRFieldDefn field(name, type);
+        const auto ogrType = toOGRFieldType(column->type(), eSubType);
+        OGRFieldDefn field(name, ogrType);
         field.SetSubType(eSubType);
+        field.SetAlternativeName(title);
+        if (width != 1 && type != ColumnType::Float && type != ColumnType::Double)
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp:269]: there's a comparison to 1 against width, and below against -1 for precision. Intended ?

Comment (bjornharrtell) [reply gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp]: Good catch, it was a typo.

**Comment by bjornharrtell** [comment]
The windows build warning is no fun to work around in my opinion (https://stackoverflow.com/questions/20919650/warning-c4800-int-forcing-value-to-bool-true-or-false-performance-warn). @rouault what's your take on it, work around or disable warning?

**Comment by rouault** [comment]
> @rouault what's your take on it, work around or disable warning?

You can use the CPL_TO_BOOL(x) macro that does ``inline static bool CPL_TO_BOOL(int x) { return x != 0; }`` . Or just test against 0

File: gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp [hunk 3]
```diff
@@ -223,8 +223,22 @@ const std::vector<Offset<Column>> OGRFlatGeobufLayer::writeColumns(FlatBufferBui
         const auto field = m_poFeatureDefn->GetFieldDefn(i);
         const auto name = field->GetNameRef();
         const auto columnType = toColumnType(field->GetType(), field->GetSubType());
+        const auto title = field->GetAlternativeNameRef();
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp:226]: GetAlternativeNameRef() never returns a nullptr, but an empty string if it is not set. I guess you should test for that, and if that's empty set the flatgeobuf field to nullptr

**Comment by bjornharrtell** [comment]
@rouault this is now ready.

**Comment by rouault** [comment]
We'd probably need a test for the new capabilities

**Comment by bjornharrtell** [comment]
@rouault you are right of course, some basic tests are now added here.

**Pull request merged**